File: CS/Scaffolding.CustomFilter/ViewModels/IssueContextViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.ComponentModel;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using Scaffolding.CustomFilter.Common.DataModel;
using Scaffolding.CustomFilter.Common.ViewModel;
using Scaffolding.CustomFilter.IssueContextDataModel;
using Scaffolding.CustomFilter.Model;

namespace Scaffolding.CustomFilter.ViewModels {
    /// <summary>
    /// Represents the root POCO view model for the IssueContext data model.
    /// </summary>
    public partial class IssueContextViewModel : DocumentsViewModel<IssueContextModuleDescription, IIssueContextUnitOfWork> {

        const string TablesGroup = "Tables";

        const string ViewsGroup = "Views";

        /// <summary>
        /// Creates a new instance of IssueContextViewModel as a POCO view model.
        /// </summary>
        public static IssueContextViewModel Create() {
            return ViewModelSource.Create(() => new IssueContextViewModel());
        }

        /// <summary>
        /// Initializes a new instance of the IssueContextViewModel class.
        /// This constructor is declared protected to avoid undesired instantiation of the IssueContextViewModel type without the POCO proxy factory.
        /// </summary>
        protected IssueContextViewModel()
            : base(UnitOfWorkSource.GetUnitOfWorkFactory()) {
        }

        protected override IssueContextModuleDescription[] CreateModules() {
            return new IssueContextModuleDescription[] {
                new IssueContextModuleDescription("Issues", "IssueCollectionView", TablesGroup, GetPeekCollectionViewModelFactory(x => x.Issues)),
            };
        }



    }

    public partial class IssueContextModuleDescription : ModuleDescription<IssueContextModuleDescription> {
        public IssueContextModuleDescription(string title, string documentType, string group, Func<IssueContextModuleDescription, object> peekCollectionViewModelFactory = null)
            : base(title, documentType, group, peekCollectionViewModelFactory) {
        }
    }
}

File: VB/T111346/ViewModels/IssueContextViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.ComponentModel
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports T111346.Common.DataModel
Imports T111346.Common.ViewModel
Imports T111346.IssueContextDataModel
Imports T111346.Scaffolding.CustomFilter.Model
Namespace Global.T111346.ViewModels
    ''' <summary>
    ''' Represents the root POCO view model for the IssueContext data model.
    ''' </summary>
    Partial Public Class IssueContextViewModel
        Inherits DocumentsViewModel(Of IssueContextModuleDescription, IIssueContextUnitOfWork)
        Private Const _TablesGroup As String = "Tables"
        Private Const _ViewsGroup As String = "Views"
        ''' <summary>
        ''' Creates a new instance of IssueContextViewModel as a POCO view model.
        ''' </summary>
        Public Shared Function Create() As IssueContextViewModel
            Return ViewModelSource.Create(Function() New IssueContextViewModel())
        End Function
        ''' <summary>
        ''' Initializes a new instance of the IssueContextViewModel class.
        ''' This constructor is declared protected to avoid undesired instantiation of the IssueContextViewModel type without the POCO proxy factory.
        ''' </summary>
        Protected Sub New()
            MyBase.New(UnitOfWorkSource.GetUnitOfWorkFactory())
        End Sub
        Protected Overrides Function CreateModules() As IssueContextModuleDescription()
            Return New IssueContextModuleDescription() {New IssueContextModuleDescription("Issues", "IssueCollectionView", _TablesGroup, GetPeekCollectionViewModelFactory(Function(x) x.Issues))}
        End Function
    End Class
    Partial Public Class IssueContextModuleDescription
        Inherits ModuleDescription(Of IssueContextModuleDescription)
        Public Sub New(ByVal title As String, ByVal documentType As String, ByVal group As String, Optional ByVal peekCollectionViewModelFactory As Func(Of IssueContextModuleDescription, Object) = Nothing)
            MyBase.New(title, documentType, group, peekCollectionViewModelFactory)
        End Sub
    End Class
End Namespace
